### ISSUE TITLE
[bug] github workflows incorrectly flags gha-pin-actions-to-sha when using local composite actions

### ISSUE THREAD

**Issue by ivml** [asamarts/alint#208]
### alint version

alint 0.15.2 (923e2398660a, built 2026-08-22)

### Platform

macOS arm64

### Minimal `.alint.yml` reproducing the issue

```yaml
version: 1
extends:
  - alint://bundled/ci/github-actions@v1
```

### MInimal reproducible example

https://github.com/ivml/alint-gha-example

### Expected output

✓ All 3 rule(s) passed.

### Actual output

```text
─── .github/workflows/test.yml ─────────────────────────────────────────────────────────────────────────────────────────
  ⚠  warning  gha-pin-actions-to-sha
              Third-party action is not pinned to a commit SHA. Pin with `@<40-char-sha> # v4.1.1` so a compromised tag
              can't silently change what runs.
              docs: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

Summary (1 violation):
  ⚠ 1 warning
  2 passing · 1 failing
```

### Description

Basically, if you reference composite actions that are available locally in the repo, `alint check` would flag that as a violation. I don't think references to local files benefit from pinning since they share the same trust boundary. Plus, it's not even possible unless you point at an older SHA (but then checkout actions would require fetch depth).

### Pre-flight checks

- [x] I searched [open + closed issues](https://github.com/asamarts/alint/issues?q=is%3Aissue) and didn't find this bug
- [x] I'm running the latest version (or have a reason not to be)

**Comment by asamarts** [comment]
Hi @ivml, thanks for reporting this, and for including a small repro. It made the fix easy to sort out.

You're right that a local ./ composite action lives in the same repo, so there's nothing to pin it to, and the rule shouldn't have flagged it.

This is fixed in #210, which is now on main. gha-pin-actions-to-sha now allows any ./ reference. The odd part is that alint's own config already had this exemption, but it had fallen out of the bundled ruleset that people extend, so we added a test to keep the two in sync from now on. We also caught the same rule flagging digest-pinned docker://...@sha256: actions, so that got fixed in the same change.

It will be in the next release (0.15.3). Thanks again for flagging it.

**Comment by asamarts** [comment]
Hey @ivml , thanks again for reporting the issue. The fix is now released in v0.16.0.
